### PR TITLE
Support AAD credential to support storage account with disabling account key

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.
Samples failed when azure login: https://github.com/microsoft/promptflow/actions/runs/9406919010/job/25911489619
Offline discuss with @crazygao, it seems related with Powershell issues and suggest removing the "enable-AzPSSession: true"

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-azure/promptflow/azure/operations/_artifact_utilities.py [hunk 1]
```diff
@@ -103,6 +110,27 @@ def get_datastore_info(operations: DatastoreOperations, name: str) -> Dict[str,
     return datastore_info
 
 
+def check_scope_rbac_role(client, scope: str, role_name: str = "Storage Blob Data Contributor"):
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/operations/_artifact_utilities.py:113]: why do we need to do such check? looks like we need to hard code a role name here. 
User can make other roles wich cover this capability?

File: src/promptflow-azure/pyproject.toml [hunk 2]
```diff
@@ -46,6 +46,7 @@ azure-ai-ml = ">=1.14.0,<2.0.0"
 azure-cosmos = ">=4.5.1,<5.0.0"  # used to upload trace to cloud
 pyjwt = ">=2.4.0,<3.0.0"  # requirement of control plane SDK
 promptflow-devkit = "<2.0.0"
+azure-mgmt-authorization = ">=1.0.0"  # used to get role assignments
```
Comment (wangchao1230) [inline src/promptflow-azure/pyproject.toml:49]: Let's not add this

File: src/promptflow-azure/promptflow/azure/operations/_fileshare_storeage_helper.py [hunk 3]
```diff
@@ -27,7 +28,38 @@
...
+        else:
+            # need add backup token intent for file share client when using AAD credential
+            self.directory_client = ShareDirectoryClient(
+                account_url=account_url,
+                credential=credential,
+                share_name=file_share_name,
+                directory_path=ARTIFACT_ORIGIN,
+                token_intent="backup",
+            )
+            self.legacy_directory_client = ShareDirectoryClient(
```
Comment (wangchao1230) [inline src/promptflow-azure/promptflow/azure/operations/_fileshare_storeage_helper.py:42]: where is this legacy_directory_client used?

Comment (YingChen1996) [reply src/promptflow-azure/promptflow/azure/operations/_fileshare_storeage_helper.py:42]: Not sure, just copy super init function here to align with sdk v2

File: src/promptflow-azure/pyproject.toml (hunk 2 above)
Comment (YingChen1996) [reply src/promptflow-azure/pyproject.toml:49]: removed

File: src/promptflow-azure/promptflow/azure/operations/_fileshare_storeage_helper.py (hunk 3 above)
Comment (YingChen1996) [reply src/promptflow-azure/promptflow/azure/operations/_fileshare_storeage_helper.py:42]: remove legacy_directory_client

**Review by wangchao1230 — APPROVED** [review]

File: docs/tutorials/index.md [hunk 4]
```diff
@@ -33,8 +33,6 @@ This section contains a collection of flow samples and step-by-step tutorials.
...
 |Rag|[Tutorial: how prompt flow helps on quality improvement](https://github.com/microsoft/promptflow/blob/main/examples/tutorials/flow-fine-tuning-evaluation/promptflow-quality-improvement.md)| This tutorial is designed to enhance your understanding of improving flow quality through prompt tuning and evaluation|
 |Rag|[Chat with pdf - test, evaluation and experimentation](https://github.com/microsoft/promptflow/blob/main/examples/flows/chat/chat-with-pdf/chat-with-pdf.ipynb)|A tutorial of chat-with-pdf flow that allows user ask questions about the content of a PDF file and get answers|
-|Rag|[Develop copilot with promptflow](https://github.com/microsoft/promptflow/blob/main/examples/tutorials/develop-copilot-with-promptflow/develop-copilot-with-promptflow.md)| In this tutorial, we will provide a detailed walkthrough on creating a RAG-based copilot using the Azure Machine Learning promptflow toolkit|
```
Comment (wangchao1230) [inline docs/tutorials/index.md:36]: why this is removed?

File: .github/workflows/samples_connections.yml [hunk 5]
```diff
@@ -78,7 +78,6 @@ jobs:
           subscription-id: ${{secrets.AZURE_SUBSCRIPTION_ID}}
           tenant-id: ${{secrets.AZURE_TENANT_ID}}
           client-id: ${{secrets.AZURE_CLIENT_ID}}
-          enable-AzPSSession: true
```
Comment (wangchao1230) [inline .github/workflows/samples_connections.yml:81]: add PR description on why this is removed.

File: docs/tutorials/index.md (hunk 4 above)
Comment (YingChen1996) [reply docs/tutorials/index.md:36]: run readme.py will remove it automatically

File: .github/workflows/samples_connections.yml (hunk 5 above)
Comment (YingChen1996) [reply .github/workflows/samples_connections.yml:81]: Samples failed when azure login: https://github.com/microsoft/promptflow/actions/runs/9406919010/job/25911489619

Offline discuss with @crazygao, maybe it related with powershell issues and remove the "enable-AzPSSession: true"

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow-devkit/tests/_constants.py [hunk 6]
```diff
@@ -8,7 +8,7 @@
 # below constants are used for pfazure and global config tests
 DEFAULT_SUBSCRIPTION_ID = "96aede12-2f73-41cb-b983-6d11a904839b"
 DEFAULT_RESOURCE_GROUP_NAME = "promptflow"
-DEFAULT_WORKSPACE_NAME = "promptflow-eastus2euap"
+DEFAULT_WORKSPACE_NAME = "promptflow-eastus"
```
Comment (wangchao1230) [inline src/promptflow-devkit/tests/_constants.py:11]: why change this?

**Review by wangchao1230 — APPROVED** [review]

**Review by lalala123123 — APPROVED** [review]

**Pull request merged**